[elasticsearch] 6.0 builds but will not run

When I checkout the latest core-plans and build a docker container from the Elasticsearch plan, it builds correctly, but fails to run. It seems the permissions on the config files are not set correctly? Here is the error:

`2017-12-05T20:02:07.911578421Z elasticsearch.default(SV): Starting service as user=hab, group=hab
2017-12-05T20:02:08.059368435Z elasticsearch.default(O): grep: /hab/pkgs/finalze/elasticsearch/6.0.0/20171205180357/es/config/jvm.options: Permission denied
2017-12-05T20:02:08.498089534Z elasticsearch.default(O): Exception in thread "main" 2017-12-05 20:02:08,496 main ERROR No log4j2 configuration file found. Using default configuration: logging only errors to the console. Set system property 'log4j2.debug' to show Log4j2 internal initialization logging.
2017-12-05T20:02:08.541876749Z elasticsearch.default(O): 2017-12-05 20:02:08,540 main ERROR Could not register mbeans java.security.AccessControlException: access denied ("javax.management.MBeanTrustPermission" "register")
2017-12-05T20:02:08.541896861Z elasticsearch.default(O):    at java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
2017-12-05T20:02:08.541899885Z elasticsearch.default(O):    at java.lang.SecurityManager.checkPermission(SecurityManager.java:585)
2017-12-05T20:02:08.541902129Z elasticsearch.default(O):    at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.checkMBeanTrustPermission(DefaultMBeanServerInterceptor.java:1848)
2017-12-05T20:02:08.541904402Z elasticsearch.default(O):    at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerMBean(DefaultMBeanServerInterceptor.java:322)
2017-12-05T20:02:08.541916630Z elasticsearch.default(O):    at com.sun.jmx.mbeanserver.JmxMBeanServer.registerMBean(JmxMBeanServer.java:522)
2017-12-05T20:02:08.541919035Z elasticsearch.default(O):    at org.apache.logging.log4j.core.jmx.Server.register(Server.java:389)
2017-12-05T20:02:08.541921115Z elasticsearch.default(O):    at org.apache.logging.log4j.core.jmx.Server.reregisterMBeansAfterReconfigure(Server.java:167)
2017-12-05T20:02:08.541923283Z elasticsearch.default(O):    at org.apache.logging.log4j.core.jmx.Server.reregisterMBeansAfterReconfigure(Server.java:140)
2017-12-05T20:02:08.541925375Z elasticsearch.default(O):    at org.apache.logging.log4j.core.LoggerContext.setConfiguration(LoggerContext.java:556)
2017-12-05T20:02:08.541927386Z elasticsearch.default(O):    at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:617)
2017-12-05T20:02:08.541929421Z elasticsearch.default(O):    at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:634)
2017-12-05T20:02:08.541931499Z elasticsearch.default(O):    at org.apache.logging.log4j.core.LoggerContext.start(LoggerContext.java:229)
2017-12-05T20:02:08.541933582Z elasticsearch.default(O):    at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:242)
2017-12-05T20:02:08.541936301Z elasticsearch.default(O):    at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:45)
2017-12-05T20:02:08.541938460Z elasticsearch.default(O):    at org.apache.logging.log4j.LogManager.getContext(LogManager.java:174)
2017-12-05T20:02:08.541940502Z elasticsearch.default(O):    at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:648)
2017-12-05T20:02:08.541942487Z elasticsearch.default(O):    at org.elasticsearch.common.logging.ESLoggerFactory.getLogger(ESLoggerFactory.java:54)
2017-12-05T20:02:08.541944533Z elasticsearch.default(O):    at org.elasticsearch.common.logging.ESLoggerFactory.getLogger(ESLoggerFactory.java:62)
2017-12-05T20:02:08.541946622Z elasticsearch.default(O):    at org.elasticsearch.common.logging.Loggers.getLogger(Loggers.java:101)
2017-12-05T20:02:08.541948675Z elasticsearch.default(O):    at org.elasticsearch.ExceptionsHelper.<clinit>(ExceptionsHelper.java:42)
2017-12-05T20:02:08.541950908Z elasticsearch.default(O):    at org.elasticsearch.ElasticsearchException.toString(ElasticsearchException.java:663)
2017-12-05T20:02:08.541952985Z elasticsearch.default(O):    at java.lang.String.valueOf(String.java:2994)
2017-12-05T20:02:08.541956887Z elasticsearch.default(O):    at java.io.PrintStream.println(PrintStream.java:821)
2017-12-05T20:02:08.541959025Z elasticsearch.default(O):    at java.lang.Throwable$WrappedPrintStream.println(Throwable.java:748)
2017-12-05T20:02:08.541961063Z elasticsearch.default(O):    at java.lang.Throwable.printStackTrace(Throwable.java:655)
2017-12-05T20:02:08.541963272Z elasticsearch.default(O):    at java.lang.Throwable.printStackTrace(Throwable.java:643)
2017-12-05T20:02:08.541965258Z elasticsearch.default(O):    at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1061)
2017-12-05T20:02:08.541967262Z elasticsearch.default(O):    at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1052)
2017-12-05T20:02:08.541970413Z elasticsearch.default(O):    at java.lang.Thread.dispatchUncaughtException(Thread.java:1959)
2017-12-05T20:02:08.541972449Z elasticsearch.default(O): 
2017-12-05T20:02:08.545984026Z elasticsearch.default(O): SettingsException[Failed to load settings from /hab/pkgs/finalze/elasticsearch/6.0.0/20171205180357/es/config/elasticsearch.yml]; nested: AccessDeniedException[/hab/pkgs/finalze/elasticsearch/6.0.0/20171205180357/es/config/elasticsearch.yml];
2017-12-05T20:02:08.546009017Z elasticsearch.default(O):    at org.elasticsearch.node.InternalSettingsPreparer.prepareEnvironment(InternalSettingsPreparer.java:101)
2017-12-05T20:02:08.546014426Z elasticsearch.default(O):    at org.elasticsearch.cli.EnvironmentAwareCommand.createEnv(EnvironmentAwareCommand.java:78)
2017-12-05T20:02:08.546018587Z elasticsearch.default(O):    at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:69)
2017-12-05T20:02:08.546023814Z elasticsearch.default(O):    at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:134)
2017-12-05T20:02:08.546027695Z elasticsearch.default(O):    at org.elasticsearch.cli.Command.main(Command.java:90)
2017-12-05T20:02:08.546100074Z elasticsearch.default(O):    at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92)
2017-12-05T20:02:08.546107145Z elasticsearch.default(O):    at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:85)
2017-12-05T20:02:08.546207798Z elasticsearch.default(O): Caused by: java.nio.file.AccessDeniedException: /hab/pkgs/finalze/elasticsearch/6.0.0/20171205180357/es/config/elasticsearch.yml
2017-12-05T20:02:08.546219262Z elasticsearch.default(O):    at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
2017-12-05T20:02:08.546233816Z elasticsearch.default(O):    at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
2017-12-05T20:02:08.546239039Z elasticsearch.default(O):    at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
2017-12-05T20:02:08.546242505Z elasticsearch.default(O):    at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
2017-12-05T20:02:08.546309382Z elasticsearch.default(O):    at java.nio.file.Files.newByteChannel(Files.java:361)
2017-12-05T20:02:08.546314457Z elasticsearch.default(O):    at java.nio.file.Files.newByteChannel(Files.java:407)
2017-12-05T20:02:08.546318031Z elasticsearch.default(O):    at java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:384)
2017-12-05T20:02:08.546322157Z elasticsearch.default(O):    at java.nio.file.Files.newInputStream(Files.java:152)
2017-12-05T20:02:08.546397787Z elasticsearch.default(O):    at org.elasticsearch.common.settings.Settings$Builder.loadFromPath(Settings.java:1058)
2017-12-05T20:02:08.546402537Z elasticsearch.default(O):    at org.elasticsearch.node.InternalSettingsPreparer.prepareEnvironment(InternalSettingsPreparer.java:99)
2017-12-05T20:02:08.546405772Z elasticsearch.default(O):    ... 6 more
2017-12-05T20:02:08.595575584Z hab-launch(SV): Child for service 'elasticsearch.default' with PID 16345 exited with code exit code: 1`

checking out revision 7a57203db328b6a5a9e201f80df05b98fd23bdbc and building 5.6.4 yields a working service in docker.

--- a/elasticsearch/config/elasticsearch.yml
+++ b/elasticsearch/config/elasticsearch.yml
@@ -26,7 +26,7 @@ node.master: {{cfg.node.master}}
 node.data: {{cfg.node.data}}
 {{/if ~}}
 
-path.logs: {{pkg.svc_var_path}}/logs
+path.logs: {{pkg.svc_var_path}}/{{cfg.path.logs}}
 path.data: {{pkg.svc_data_path}}/{{cfg.path.data}}
 
 indices.recovery.max_bytes_per_sec: {{cfg.indices.recovery.max_bytes_per_sec}}

--- a/elasticsearch/hooks/run
+++ b/elasticsearch/hooks/run
@@ -13,4 +13,5 @@ ulimit -n {{cfg.runtime.max_open_files}}
 {{/if ~}}
 
 export ES_JVM_OPTIONS="{{pkg.svc_config_path}}/jvm.options"
-exec elasticsearch -Epath.conf={{pkg.svc_config_path}}
+export ES_PATH_CONF="{{pkg.svc_config_path}}"
+exec elasticsearch

--- a/elasticsearch/hooks/run
+++ b/elasticsearch/hooks/run
@@ -12,6 +12,5 @@ ulimit -l {{cfg.runtime.max_locked_memory}}
 ulimit -n {{cfg.runtime.max_open_files}}
 {{/if ~}}
 
-export ES_JVM_OPTIONS="{{pkg.svc_config_path}}/jvm.options"
 export ES_PATH_CONF="{{pkg.svc_config_path}}"
 exec elasticsearch

--- a/elasticsearch/plan.sh
+++ b/elasticsearch/plan.sh
@@ -38,7 +38,7 @@ do_install() {
   cp -a ./* "${pkg_prefix}/es"
 
   # jvm.options needs to live relative to the binary.
-  # mkdir -p mkdir -p "$pkg_prefix/es/config"
+  # mkdir -p "$pkg_prefix/es/config"
   # install -vDm644 config/jvm.options "$pkg_prefix/es/config/jvm.options"
 
   # Delete unused binaries to save space

